gui/src/2D/controls/selector.ts

```diff
@@ -0,0 +1,528 @@
+import {Rectangle} from "./rectangle";
+import {StackPanel} from "./stackPanel";
+import {Control} from "./control";
+import {TextBlock} from "./textBlock";
+import {Checkbox} from "./checkbox";
+import {RadioButton} from "./radioButton";
+import {Slider} from "./slider";
+
+/** Class used to create a SelectorGroup 
+ * which contains groups of checkboxes, radio buttons and sliders
+*/
+export class SelectorGroup {
+    private _selectNb = 0;
+    private _groupPanel = new StackPanel();
+    private _selectors: StackPanel[] = new Array();
+    private _sliders: Slider[] = new Array();
+    
+
+    /**
+     * Creates a new SelectorGroup
+     * @param name of group, used as a group heading
+     * @param type specifies a check box, radio button or slider grouping
+     */
+    constructor(public name: string, public type: string) {
+        if (type === void 0) { type = "C"; }
+        type = type.substr(0,1).toUpperCase();
+        if(type !=="R") {
+            if(type !== "S") {
+                if(type !== "C") {
+                    type = "C";
+                }
+            }
+        }
+        this._groupPanel.verticalAlignment = Control.VERTICAL_ALIGNMENT_TOP;
+        this._groupPanel.horizontalAlignment = Control.HORIZONTAL_ALIGNMENT_LEFT;
+        this._addGroupHeader(name);       
+    }
+
+    /** Gets group stackPanel  */
+    public get group(): StackPanel {
+        return this._groupPanel;
+    }
+
+    /** Gets selectors  */
+    public get selectors(): StackPanel[] {
+        return this._selectors;
+    }
+
+    /** Gets siders  */
+    public get sliders(): Slider[] {
+        return this._sliders;
+    }
+
+    /** Adds a checkbox or radio button to the SelectorGroup
+     * @param label is the label for the selector
+     * @param func is the function called when the Selector is checked
+     * @param checked is true when Selector is checked
+     */
+    public addSelector(label: string, func?: () => any , checked?: boolean): void {
+        if(this.type === "S") {
+            return
+        }
+        if (func === void 0) { func = function(){}; }

```

Comment:
I would recommend using TS construct for that:
public addSelector(label: string, func: () => any = {} , checked = false): void {

Reply:
It should :)